sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotation.kt

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.maps.viewannotation
+
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.FrameLayout
+import com.mapbox.maps.ViewAnnotationOptions
+import java.util.*
+
+internal data class ViewAnnotation(
+  /**
+   * Inflated Android view.
+   */
+  val view: View,
+  /**
+   * We handle visibility automatically in the manager
+   * if user did not specify [ViewAnnotationOptions.visible]  explicitly  during add / update operation
+   */
+  var handleVisibilityAutomatically: Boolean,
+  /**
+   * Cached value if view is visible before OnGlobalLayoutListener was triggered.
+   */
+  var visible: Boolean,
+  /**
+   * Layout params needed to position Android view correctly on the screen.
+   */
+  var viewLayoutParams: FrameLayout.LayoutParams,
+) {
+  /**
+   * String id needed to call functions from gl-native
+   */
+  val id: String = (VIEW_ANNOTATION_CURRENT_ID++).toString()
+
+  /**
+   * Global layout listener to control positioning on the screen based on view's actual visibility.
+   */
+  var globalLayoutListener: ViewTreeObserver.OnGlobalLayoutListener? = null
```

Comment:
do we need a GlobalLayoutListener for each annotation independently? 
why not have 1 in the manager that manages all the annotations separately.

Reply:
that's a valid point, I'll create follow-up task about it, changing it now may be a bit risky

Reply:
https://github.com/mapbox/mapbox-maps-android/issues/862